lib/dmv.rb

```diff
@@ -9,8 +13,166 @@ def add_facility(facility)
+    vehicle_year_total = 0
+    @facilities.each do |facility|
+      if facility.state == state
+        vehicle_year_total += facility.registered_vehicles.count do |vehicle|
+          vehicle.registration_date.year == specified_year
+        end
+      end
+    end
+
+    #Find county which has the most vehicle registrations.  Note high machinery overlap with most popular model...a way to combine?
+    county_tally = {}
+    @facilities.each do |facility|
+      if facility.state == state
+        facility.registered_vehicles.each do |vehicle|
+          if county_tally.include?(vehicle.registration_county)
+            county_tally[vehicle.registration_county] += 1
+          else
+            county_tally[vehicle.registration_county] = 1
+          end
+        end
+      end
+    end
+    most_popular_county = county_tally.key(county_tally.values.max)
+
+    {most_popular_model: most_popular_model, number_registered_for_year: vehicle_year_total, county_most_registered_vehicles: most_popular_county}
+  end
+
+  def facility_parameters_co(facility_co_data)
+    #Try to centralize idiosyncratic aspects of each state's dataset to specific spots in codebase
+    #Alternate: could hosue this in Facility class.  Might try that another time - would require a different format since Facility wouldn't yet be initialized
+    {
+      name: facility_co_data[:dmv_office],
+      address: "#{facility_co_data[:address_li]} #{facility_co_data[:address__1]} #{facility_co_data[:location]} #{facility_co_data[:city]} #{facility_co_data[:state]} #{facility_co_data[:zip]}",
+      phone: facility_co_data[:phone],
+      hours: facility_co_data[:hours]       #Iteration 4: add hours (direct copy for CO)
+    }
+  end
+
+  def facility_parameters_ny(facility_ny_data)
+    #To format name: have to concatenate and then 'fix' capitlization the hard way (since each word is capitalized for a title usually)
+    name_formatted = facility_ny_data[:office_name].split.map { |word| word.capitalize }
+    name_formatted << (facility_ny_data[:office_type].split.map { |word| word.capitalize }).join(" ")
+    name_formatted = name_formatted.join(" ")
+
+    #To format address: lots of overlap here with previous entry.  If I knew how to make a method that accepted varying arguments, I might be able to; don't know this yet...
+    address_formatted = facility_ny_data[:street_address_line_1].split.map { |word| word.capitalize }
+    address_formatted << facility_ny_data[:city].split.map { |word| word.capitalize }
+    address_formatted << facility_ny_data[:state]
+    address_formatted << facility_ny_data[:zip_code]
+    address_formatted = address_formatted.join(" ")
+
+    #Hours are listed for each day by opening and closing.  For now, I'm going to brute-force this,
+    #since no exact format is expected.  This is a mess though:
+    #This is gross...some facilities don't have keys for all M-F.  So I need to run ifs now (otherwise problems with nil)
+    hours_formatted = ""
```

Comment:
This data cleaning does get a bit icky. That's why extracting into a helper method that can be called such as `format_opening_times` could at least encapsulate the ickyness.